### PR TITLE
Realtime Channel history

### PR DESCRIPTION
**Pull request by tiholic**
Realtime#Channel#Channel#history will return PaginatedResult.

```dart
void getHistory([ably.RealtimeHistoryParams params]) async {
    var result = await channel.history(params);

    var messages = result.items; //get messages
    var hasNextPage = result.hasNext(); //tells whether there are more results
    if(hasNextPage){    
      result = await result.next();  //fetches next page results
      messages = result.items;
    }
    if(!hasNextPage){
      result = await result.first();  //fetches first page results
      messages = result.items;
    }
}

// history with default params
getHistory();

// sorted and filtered history
getHistory(
  ably.RealtimeHistoryParams(
    direction: 'forwards',
    limit: 10,
    start: DateTime.fromMillisecondsSinceEpoch(0),
    end: DateTime.now(),
    untilAttach: false
  )
);
```

### CODE REVIEW

**Review by zoechi — APPROVED** [review]

File: example/lib/main.dart [hunk 1]
```diff
@@ -447,44 +448,101 @@ class _MyAppState extends State<MyApp> {
...
+                    _restHistory = await _restHistory.next();
+                  } else {
+                    _restHistory = await _restHistory.first();
+                  }
+                  setState(() {});
+                } on ably.AblyException catch (e) {
+                  print('failed to get history:: $e :: ${e.errorInfo}');
+                }
+              },
+        onLongPress: (_rest == null)
```
Comment (QuintinWillison) [inline example/lib/main.dart:493]: Interesting. Is this behaviour obvious? Maybe I don't fully understand the context (I've not loaded the app in a while, least not for this branch), but I would usually expect a long-press to first present a menu or something.

File: test_integration/lib/test/realtime_history_test.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    final historyLimit2 =
+        await _history(channel, RealtimeHistoryParams(limit: 2));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final historyForwardLimit4 = await _history(
+        channel, RealtimeHistoryParams(direction: 'forwards', limit: 4));
+    await Future.delayed(const Duration(seconds: 2));
+
+    final time1 = DateTime.now();
+    //TODO(tiholic) iOS fails without this delay
```
Comment (QuintinWillison) [inline test_integration/lib/test/realtime_history_test.dart:51]: [ditto](https://github.com/ably/ably-flutter/pull/68#discussion_r576150484) 😁

File: test_integration/lib/test/test_names.dart [hunk 3]
```diff
@@ -19,6 +19,8 @@ class TestName {
   static const String realtimePublish = 'realtimePublish';
   static const String realtimeEvents = 'realtimeEvents';
   static const String realtimeSubscribe = 'realtimeSubscribe';
+  static const String realtimeHistory = 'realtimeHistory';
   static const String realtimePublishWithAuthCallback =
       'realtimePublishWithAuthCallback';
+  // TODO(tiholic) handle realtimeHistoryWithAuthCallback
```
Comment (QuintinWillison) [inline test_integration/lib/test/test_names.dart:25]: [ditto](https://github.com/ably/ably-flutter/pull/68#discussion_r576150798) 😁

**Review by QuintinWillison — APPROVED** [review]

File: example/lib/main.dart (hunk 1 above)
Comment (tiholic) [reply example/lib/main.dart:493]: @QuintinWillison long press is not intuitive, I had a plan to enhance this, but primary focus was not on example.

File: test_integration/lib/test/realtime_history_test.dart (hunk 2 above)
Comment (tiholic) [reply test_integration/lib/test/realtime_history_test.dart:51]: https://github.com/ably/ably-flutter/issues/86

File: test_integration/lib/test/test_names.dart (hunk 3 above)
Comment (tiholic) [reply test_integration/lib/test/test_names.dart:25]: https://github.com/ably/ably-flutter/issues/87

**Pull request merged**